### PR TITLE
Admin Delete Environment action

### PR DESCRIPTION
**Pull request by aditipuranik**
Admin Delete Action

This will delete an envioronment.
Requires Environment URL, username and password

### CODE REVIEW

**Review by swapnilbhalgat — APPROVED** [review]

File: src/actions/delete-environment/index.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        const envUrl = core.getInput('environment-url', { required: true });
+        const username = core.getInput('user-name', { required: true });
+        core.info(`environmentUrl: ${envUrl}; login as user: ${username}`);
+
+        const password = core.getInput('password-secret', { required: true });
+        if (!password || password.length === 0) {
+            return core.setFailed('Missing password! Specify one by setting input: \'password-secret\'');
+        }
+
+        const performAsync = getInputAsBool('async', false, false);
```
Comment (davidjenni) [inline src/actions/delete-environment/index.ts:26]: remove, not defined in action,yml, and shouldn't be: pac admin delete internally does the polling until deleted. From the GH actions perspective, we don't want/need the async *interactive* behavior, the blocking/sync interactive behavior is actually desired here

File: src/actions/delete-environment/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        const pac = factory.getRunner('pac', process.cwd());
+        await pac.run(['auth', 'clear']);
+        await pac.run(['auth', 'create', '--url', envUrl, '--username', username, '--password', password]);
```
Comment (davidjenni) [inline src/actions/delete-environment/index.ts:30]: won't work, the authN profile needs to be of --kind ADMIN ...

**Review by davidjenni — CHANGES_REQUESTED** [review]

File: src/actions/delete-environment/index.ts (hunk 1 above)
Comment (aditipuranik) [reply src/actions/delete-environment/index.ts:26]: makes sense. Removed.

File: src/actions/delete-environment/index.ts (hunk 2 above)
Comment (aditipuranik) [reply src/actions/delete-environment/index.ts]: made the change

File: .vscode/delete-environment/action.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (swapnilbhalgat) [inline .vscode/delete-environment/action.yaml:1]: this folder should be under root directory

**Review by davidjenni — CHANGES_REQUESTED** [review]

**Review by swapnilbhalgat — APPROVED** [review]

File: src/actions/delete-environment/index.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import * as core from '@actions/core';
+import { DefaultRunnerFactory, ActionLogger, getInputAsBool, getWorkingDirectory, PacRunner, RunnerFactory} from '../../lib';
```
[the remark this answers is not in the document]
Comment (aditipuranik) [reply src/actions/delete-environment/index.ts]: done

File: src/actions/delete-environment/index.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            return core.setFailed('Missing password! Specify one by setting input: \'password-secret\'');
+        }
+
+        const pac = factory.getRunner('pac', process.cwd());
+        await pac.run(['auth', 'clear']);
+        pac.run(['auth', 'create', '--kind', 'ADMIN', '--username', username, '--password', password]);
+
+        const deleteEnvArgs = ['admin', 'delete', '--url', envUrl];
+        await pac.run(deleteEnvArgs);
+        core.info('delete environment');
```
[the remark this answers is not in the document]
Comment (aditipuranik) [reply src/actions/delete-environment/index.ts]: changed wording to "environment deleted"

**Review by davidjenni — APPROVED** [review]

**Review by davidjenni — APPROVED** [review]

**Pull request merged**